### PR TITLE
katello-backup test exit status

### PR DESCRIPTION
**Pull request by pondrejk**
Automating https://bugzilla.redhat.com/show_bug.cgi?id=1323607
Related issue #4607
Please also note the 6.2 cherry-pick

### CODE REVIEW

File: tests/foreman/sys/test_hot_backup.py [hunk 1]
```diff
@@ -128,6 +128,37 @@ def test_positive_online_backup_with_directory_created(self):
...
+        :bz: 1323607
+
+        :expectedresults: katello-backup finished with correct exit code
+
+        """
+        with get_connection() as connection:
+            dir_name = gen_string('alpha')
+            dead_service = 'postgresql'
+            connection.run('service {0} stop'.format(dead_service))
+            connection.run('rm -rf /tmp/{0}'.format(dir_name))
```
Comment (ldjebran) [inline tests/foreman/sys/test_hot_backup.py]: perhaps use the same cleanup function everywhere: tmp_directory_cleanup(connection, dir_name)

**Comment by ldjebran** [comment]
@pondrejk can you please provide a portion of the test log where the fail occur (of course running without the skip decorator)

**Comment by pondrejk** [comment]
@ldjebran thanks, addressed the cleanup note. Output from unskpped test:

```nosetests -v tests/foreman/sys/test_hot_backup.py:HotBackupTestCase.test_positive_online_backup_exit_code_on_failure 
katello-backup --online-backup correct exit code on failure ... FAIL

======================================================================
FAIL: katello-backup --online-backup correct exit code on failure
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pondrejk/Documents/robottelo/tests/foreman/sys/test_hot_backup.py", line 157, in test_positive_online_backup_exit_code_on_failure
    self.assertNotEqual(result.return_code, 0)
AssertionError: 0 == 0
-------------------- >> begin captured logging << --------------------
robottelo: DEBUG: Started Test: HotBackupTestCase/test_positive_online_backup_exit_code_on_failure
robottelo.ssh: DEBUG: Instantiated Paramiko client 0x7f7455da1290
robottelo.ssh: INFO: Connected to [None]
robottelo.ssh: INFO: >>> service postgresql stop
robottelo.ssh: INFO: <<< stderr
Redirecting to /bin/systemctl stop  postgresql.service

robottelo.ssh: INFO: >>> rm -rf /tmp/kGmalzKMNY
robottelo.ssh: INFO: >>> katello-backup /tmp/kGmalzKMNY --online-backup
robottelo.ssh: INFO: <<< stdout
Starting backup: 2017-04-28 08:12:14 -0400
Creating backup folder /tmp/kGmalzKMNY/katello-backup-2017-04-28T08:12:14-04:00
Generating metadata ... 
Backing up config files... 
Done.
Backing up postgres db... 
Done.
Backing up mongo db... 
Done.
Backing up Pulp data... 
Done.
Done with backup: 2017-04-28 08:12:20 -0400
**** BACKUP Complete, contents can be found in: /tmp/kGmalzKMNY/katello-backup-2017-04-28T08:12:14-04:00 ****

robottelo.ssh: INFO: <<< stderr
rake aborted!
PG::Error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:651:in `initialize'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:651:in `new'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:651:in `connect'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:242:in `initialize'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:44:in `new'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql_adapter.rb:44:in `postgresql_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:438:in `new_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:448:in `checkout_new_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:422:in `acquire_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:349:in `block in checkout'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:348:in `checkout'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:263:in `block in connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:262:in `connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:571:in `retrieve_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_handling.rb:113:in `retrieve_connection'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activerecord-4.2.6/lib/active_record/connection_handling.rb:87:in `connection'
/opt/theforeman/tfm/root/usr/share/gems/gems/ancestry-2.2.1/lib/ancestry/has_ancestry.rb:46:in `has_ancestry'
/usr/share/foreman/app/models/concerns/nested_ancestry_common.rb:7:in `block in <module:NestedAncestryCommon>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/concern.rb:120:in `class_eval'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/concern.rb:120:in `append_features'
/usr/share/foreman/app/models/taxonomy.rb:5:in `include'
/usr/share/foreman/app/models/taxonomy.rb:5:in `<class:Taxonomy>'
/usr/share/foreman/app/models/taxonomy.rb:1:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:360:in `require_or_load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:526:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:526:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/usr/share/foreman/app/models/nic/base.rb:40:in `<class:Base>'
/usr/share/foreman/app/models/nic/base.rb:4:in `<module:Nic>'
/usr/share/foreman/app/models/nic/base.rb:3:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:360:in `require_or_load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/usr/share/foreman/app/models/nic/interface.rb:2:in `<module:Nic>'
/usr/share/foreman/app/models/nic/interface.rb:1:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:360:in `require_or_load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/usr/share/foreman/app/models/nic/managed.rb:2:in `<module:Nic>'
/usr/share/foreman/app/models/nic/managed.rb:1:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:360:in `require_or_load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/usr/share/foreman/app/services/interface_type_mapper.rb:4:in `<class:InterfaceTypeMapper>'
/usr/share/foreman/app/services/interface_type_mapper.rb:1:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:360:in `require_or_load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/opt/theforeman/tfm/root/usr/share/gems/gems/foreman_hooks-0.3.13/lib/foreman_hooks.rb:90:in `load_missing_constant_with_hooks'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:184:in `const_missing'
/usr/share/foreman/config/initializers/apipie.rb:22:in `block in <top (required)>'
/opt/theforeman/tfm/root/usr/share/gems/gems/apipie-rails-0.4.0/lib/apipie/apipie_module.rb:22:in `configure'
/usr/share/foreman/config/initializers/apipie.rb:3:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `block in load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `load'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/engine.rb:652:in `block in load_config_initializer'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/notifications.rb:166:in `instrument'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/engine.rb:651:in `load_config_initializer'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/engine.rb:616:in `block (2 levels) in <class:Engine>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/engine.rb:615:in `each'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/engine.rb:615:in `block in <class:Engine>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:30:in `instance_exec'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:30:in `run'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:55:in `block in run_initializers'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:44:in `each'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:44:in `tsort_each_child'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/initializable.rb:54:in `run_initializers'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/application.rb:352:in `initialize!'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/railtie.rb:194:in `public_send'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/railtie.rb:194:in `method_missing'
/usr/share/foreman/config/environment.rb:5:in `<top (required)>'
/opt/rh/rh-ror42/root/usr/share/gems/gems/polyglot-0.3.4/lib/polyglot.rb:65:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/opt/rh/rh-ror42/root/usr/share/gems/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/application.rb:328:in `require_environment!'
/opt/rh/rh-ror42/root/usr/share/gems/gems/railties-4.2.6/lib/rails/application.rb:457:in `block in run_tasks_blocks'
Tasks: TOP => plugin:list => environment
(See full trace by running task with --trace)
pg_dump: [archiver (db)] connection to database "foreman" failed: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
pg_dump: [archiver (db)] connection to database "candlepin" failed: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
tar: Removing leading `/' from member names

robottelo.ssh: DEBUG: Destroyed Paramiko client 0x7f7455da1290
robottelo: DEBUG: Finished Test: HotBackupTestCase/test_positive_online_backup_exit_code_on_failure
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 22.179s

FAILED (failures=1)
```

**Comment by sghai** [comment]
ACK

**Pull request merged**